### DISCUSSION TITLE
CORS error

### DISCUSSION THREAD

**Discussion by dp100392vim** [vercel/next.js#50229] [Help] [answered]
### Summary

Hi! I have the problem... 
how to allow access from another resource

Access to fetch at 'https://my-host.vercel.app/' from origin 'https://mysite.myshopify.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

That dont work:

vercel.json
{
  "headers": [
    {
      "source": "/api/(.*)",
      "headers": [
        { "key": "Access-Control-Allow-Credentials", "value": "true" },
        { "key": "Access-Control-Allow-Origin", "value": "*" },
        { "key": "Access-Control-Allow-Methods", "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
        { "key": "Access-Control-Allow-Headers", "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version" }
      ]
    }
  ]
}

**Comment by devjiwonchoi** [comment, marked as answer]
It is stated in the [docs](https://nextjs.org/docs/app/building-your-application/routing/router-handlers#cors),

> You can set CORS headers on a Response using the standard Web API methods:

Example:
```
// app/api/route.ts

export async function GET(request: Request) {
  return new Response('Hello, Next.js!', {
    status: 200,
    headers: {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
      'Access-Control-Allow-Headers': 'Content-Type, Authorization',
    },
  });
}
```
